echopype/consolidate/api.py

```diff
@@ -191,10 +194,35 @@ def sel_interp(var, time_dim_name):
     if "longitude" not in echodata["Platform"] or echodata["Platform"]["longitude"].isnull().all():
         raise ValueError("Coordinate variables not present or all nan")
 
+    # Check if any latitude/longitude value is NaN/0
+    contains_nan_lat_lon = (
+        np.isnan(echodata["Platform"]["latitude"].values).any()
+        or np.isnan(echodata["Platform"]["longitude"].values).any()
+    )
+    contains_zero_lat_lon = (echodata["Platform"]["latitude"].values == 0).any() or (
+        echodata["Platform"]["longitude"].values == 0
+    ).any()
+    interp_msg = "Interpolation may be negatively impacted, so the user should handle these values."
+    if contains_nan_lat_lon:
+        logger.warning(f"Latitude and/or longitude arrays contain NaNs. {interp_msg}")
+    if contains_zero_lat_lon:
+        logger.warning(f"Latitude and/or longitude arrays contain zeros. {interp_msg}")
+
     interp_ds = ds.copy()
     time_dim_name = list(echodata["Platform"]["longitude"].dims)[0]
+
+    # Check if there are duplicates in time_dim_name
+    if len(np.unique(echodata["Platform"][time_dim_name].data)) != len(
+        echodata["Platform"][time_dim_name].data
+    ):
+        logger.warning(
+            f'The echodata["Platform"]["{time_dim_name}"] array contains duplicate values. '
+            "Interpolation expects unique time values."
+        )
+
```

Comment:
I think this should raise an error instead of a warning? Because the underlying code would actually raise a warning. This is as opposed to the GPS NaN/0 cases that the underlying code would run through without erroring out.

Reply:
Ah, I see. I think I misinterpreted your original statement here: https://github.com/OSOceanAcoustics/echopype/issues/1294#issuecomment-2037745091. I assumed that meant letting Pandas call the error and we provide a warning message prior to showing the user where the error occurred

Reply:
Oh I see, no problem!